drivers/console/jailhouse_debug_console.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022 NXP
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <sys/printk.h>
+#include <device.h>
+#include <init.h>
+
+#if defined(CONFIG_PRINTK) || defined(CONFIG_STDOUT_CONSOLE)
+/**
+ * @brief Output one character to SIMULATOR console
+ * @param c Character to output
+ * @return The character passed as input.
+ */
+static int console_out(int c)
+{
+	register unsigned long x0 __asm__("x0") = 8;
+	register unsigned long x1 __asm__("x1") = c;
+
+	__asm__ volatile ("hvc #0x4a48\r\n"
+			  : "+r" (x0), "+r" (x1) : : );
+	return c;
```

Comment:
Can we use `arm_smccc_hvc` instead?

Reply:
No. arm_smccc_hvc use #0 when invoking hvc/smc, but jailhouse use value 0x4a48.